tests/mosaic/profiler_cupti_test.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2024 The JAX Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for Mosaic GPU CUPTI-based profiler."""
+
+from absl.testing import absltest, parameterized
+import jax
+from jax._src import config
+from jax._src import test_util as jtu
+import jax.numpy as jnp
+try:
+  import jax._src.lib.mosaic_gpu  # noqa: F401
+  HAS_MOSAIC_GPU = True
+except ImportError:
+  HAS_MOSAIC_GPU = False
+else:
+  from jax.experimental.mosaic.gpu import profiler
+
+
+# ruff: noqa: F405
+# pylint: disable=g-complex-comprehension
+config.parse_flags_with_absl()
+
+class ProfilerCuptiTest(parameterized.TestCase):
```

Comment:
It might be better to do it in a follow up, but we probably need a single profiler test which would be parameterized over `mode`.

Reply:
The two `mode`s are fundamentally different in a pretty significant way (see `measure` docstring), so it felt more natural to write dedicated test cases.